File: 03-strings/03_longest_unique_substr.c
#include "../test_runner.h"

int length_of_longest_substring(char* s) {
    return -1;
}

int main() {
    TEST_INT("abcabcbb", length_of_longest_substring("abcabcbb"), 3);
    TEST_INT("bbbbb",    length_of_longest_substring("bbbbb"),    1);
    TEST_INT("pwwkew",   length_of_longest_substring("pwwkew"),   3);
    TEST_INT("empty",    length_of_longest_substring(""),         0);
    TEST_INT("dvdf",     length_of_longest_substring("dvdf"),     3);
    TEST_SUMMARY();
}

File: 03-strings/03_longest_unique_substr.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Longest Substring Without Repeating Characters (LC #3) -- Medium
# Find length of longest substring with all unique characters.
#
#   Key: Sliding window + hashset. Move left pointer when duplicate found.

def length_of_longest_substring(s):
    pass


run_tests(length_of_longest_substring, [
    (("abcabcbb",), 3),
    (("bbbbb",),    1),
    (("pwwkew",),   3),
    (("",),         0),
    (("au",),       2),
    (("dvdf",),     3),
])
